### DISCUSSION TITLE
blog/2024/08/18/Huawei-Matebook-e-2022-install-archlinux-double-system/

### DISCUSSION THREAD

**Discussion by giscus** [Vanilla-Yukirin/BlogComments#15] [Announcements]
# blog/2024/08/18/Huawei-Matebook-e-2022-install-archlinux-double-system/

本文详细介绍了在华为Matebook e 2022上安装Arch Linux双系统的全过程，包括准备工作、系统安装、网络配置、驱动安装等关键步骤。文章从BIOS设置入手，逐步指导读者如何配置网络，分区，安装系统核心组件，以及如何设置启动管理器GRUB。此外，还涵盖了系统优化、桌面环境安装和必要的后续配置，确保用户能够顺利完成安装并享受Arch Linux。

https://vanilla-chan.cn/blog/2024/08/18/Huawei-Matebook-e-2022-install-archlinux-double-system/

**Comment by RainbowRain9** [comment]
请问按照这个流程我可以安装bliss os吗

**Reply by Vanilla-Yukirin** [reply]
双系统的前置操作虽然类似，但是之后的流程就截然不同了。最好还是照着bliss os的安装教程比较好。这篇随笔是针对安装arch的

**Reply by RainbowRain9** [reply]
装过一次，有bug，华为 MateBook E 2022 安装了一个 Android x86（Bliss OS 16.x），然后屏幕一半显示异常，具体表现为上下抽动，触摸没有问题，但抽搐这一半块屏幕响应的点选按钮也在正常位置（也就是说仅图像抽动而控件在原位）。
我装的是双系统，当回到 Windows 11 时屏幕完全没有问题，但是就是 Bliss OS 有这个问题，官方自带的驱动也尝试过了，加载后并不能解决问题。
可能是驱动或是显卡的问题，不知道按您这个流程能不能解决

**Reply by Vanilla-Yukirin** [reply]
你提到的“屏幕一半显示异常，具体表现为上下抽动”似乎我也遇到了。我的表现像是“命令行界面被某个东西横向的截掉了一部分，然后卡住了，似乎实际上在正常运行但是画面卡死”，原因与显卡驱动有关。

我需要在安装系统的时候加入`nomodeset=0`方可进入系统，然后再更新显卡驱动，就可以正常的（不需要`nomodeset=0`）进入系统。但是我也不确定，网络上也有人出现了这个问题（matebook E 安装 arch ）：https://bbs.archlinuxcn.org/viewtopic.php?id=12223， 不知道是否相关

**Reply by RainbowRain9** [reply]
好的谢谢

**Comment by Yohane-Mashiro** [comment]
问一下现在显卡还有问题吗，上一次我安装的时候就有这个显卡问题

**Reply by Vanilla-Yukirin** [reply]
很久没有配置过了，这几年来也没有查询新资料了喵，sorry~

**Comment by Yohane-Mashiro** [comment]
最新的6.19内核依旧有问题（

**Reply by WangWindow** [reply]
可以用 6.12 lts 内核。
之后的内核好像都不行。

**Reply by v8wzge** [reply]
除了6.1.4，其他的都只能nomodeset

**Reply by v8wzge** [reply]
7.0.rc4都试了，照样不行

**Reply by WangWindow** [reply]
@v8wzge 
我的情况是如果用 6.12 以前的版本，只有安装时需要设置 nomodeset=0 启动参数，安装好重启就不用设置这个参数，直接能进入桌面，核显也能被正常识别。
如果使用大于 6.12 的内核，安装完系统重启就会黑屏，设置 nomodeset=0 启动后显卡信息是 llvmpipe，没有识别到核显。
我是用 archinstall 安装系统的。

**Reply by v8wzge** [reply]
设备看起来有差异，早在6字头内核还没出我就入手这板子想着刷Linux了，但直到去年才看到锁6.1.4的不闪屏方案，后来放弃了，之后每次内核大更新我都试下，直到看到这篇文章，我用6.18lts试试不行，看到你的评论，又切6.12lts也不行，同时我又切6.6lts和6.1lts，最新的内核都不行，6.1lts内核最新是6.1.8，左右两半闪屏，去arch缓存找那个6.1.4.arch1-1就好了

**Reply by v8wzge** [reply]
对了桌面是kde或者gnome，都不行
